docs/source/release_notes/release-notes.rst

```diff
@@ -50,6 +50,10 @@ develop
     *    - Type
          - Change
 
+    *    - |improved|
+         - Introduced runtime checks on client side for timestamps retrieved from timelock; aiming to prevent data corruption in case of timestamps going back in time, possibly caused by a misconducted timelock migration. This is a best effort for catching abnormalities on timestamps at runtime, and do not provide absolute protection.
```

Comment:
A bit of pedantry/grammar-related nits. I'd suggest rewriting this a little to split the sentences, though non-blocking:

Introduced runtime checks on _the_ client side for timestamps retrieved from timelock. _This aims_ to prevent data corruption _if_ timestamps _go_ back in time, possibly caused by a misconducted timelock migration. This is a best effort for catching abnormalities on timestamps at runtime, and _does_ not provide absolute protection.